cgi-bin/LJ/Talk.pm

```diff
@@ -3695,7 +3695,13 @@ sub make_preview {
             unless $_ eq 'body' || $_ eq 'subject' || $_ eq 'prop_opt_preformatted' || $_ eq 'editreason';
     }
 
-    $ret .= "<br /><input type='submit' value='$BML::ML{'/talkpost_do.bml.preview.postcomment'}' />\n";
+    my $post_disabled = $u->does_not_allow_comments_from($remote)
+        || ($u->{'opt_whocanreply'} eq 'reg' && $remote->is_identity && ! ( $remote->is_validated || $u->trusts( $remote )));
```

Comment:
This logic only depends on `$u` and `$remote`. Instead of having it in two places, can we move it into a method on `LJ::User`? Could we just add it to `does_not_allow_comments_from` or does that break some other code? That method _sounds_ like it should just have this behavior, since `opt_whocanreply` is just specifying who is allowed to comment...